lib/main_common.pm

```diff
@@ -2791,22 +2791,26 @@ sub load_publiccloud_tests {
     if (get_var('PUBLIC_CLOUD_PREPARE_TOOLS')) {
         loadtest "publiccloud/prepare_tools";
     }
-    elsif (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
+    elsif (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS') && !get_var('PUBLIC_CLOUD_QAM')) {
         loadtest "publiccloud/img_proof";
     }
     elsif (get_var('PUBLIC_CLOUD_QAM')) {
         loadtest "publiccloud/download_repos";
         my $args = OpenQA::Test::RunArgs->new();
-        loadtest "publiccloud/ssh_interactive_init",  run_args => $args;
-        loadtest "publiccloud/register_system",       run_args => $args;
-        loadtest "publiccloud/transfer_repos",        run_args => $args;
-        loadtest "publiccloud/patch_and_reboot",      run_args => $args;
-        loadtest "publiccloud/ssh_interactive_start", run_args => $args;
-        loadtest "publiccloud/instance_overview";
-        load_extra_tests_prepare();
-        load_extra_tests_docker() if (get_var('PUBLIC_CLOUD_CONTAINERS'));
-        load_extra_tests_console() unless (get_var('PUBLIC_CLOUD_CONTAINERS'));
-        loadtest "publiccloud/ssh_interactive_end", run_args => $args;
+        loadtest "publiccloud/ssh_interactive_init", run_args => $args;
+        loadtest "publiccloud/register_system",      run_args => $args;
+        loadtest "publiccloud/transfer_repos",       run_args => $args;
+        loadtest "publiccloud/patch_and_reboot",     run_args => $args;
+        if (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
+            loadtest("publiccloud/img_proof", run_args => $args);
```

Comment:
I'm missing the `$provider->cleanup()` in that flow. 
It's not a big deal, as we still have hopefully set `openqa_ttl` tag for the instance. But I would prefer to have some cleanup in the test and only use **pcw** for leftover deletion.